contrib/profiling.py

```diff
@@ -0,0 +1,29 @@
+from modmesh.onedim import euler1d
+from _modmesh import stop_watch
+
+
+shock_tube = euler1d.ShockTube()
+shock_tube.build_constant(
+    gamma=1.4,
+    pressure1=1.0,
+    density1=1.0,
+    pressure5=0.1,
+    density5=0.125,
+)
+
+shock_tube.build_numerical(xmin=-1, xmax=1, ncoord=21, time_increment=0.05)
+
+# reset lap
```

Comment:
Is there a report or something for the test case?

Reply:
No. We can (and should) implement that for sure. 

Do we want to add profiling a part of test case? If yes, then we may want to have a benchmark that is agreed on mutually.

Reply:
Before making automatic tests for runtime, perhaps we can first work on the (automatic generation of the) profiling reports.  The reports are what we want from a profiler.  One example of the reporting is https://github.com/yse/easy_profiler .

However here I think we should first develop a [scoped-based profiler](https://yyc.solvcon.net/en/latest/nsd/11arraydesign/arraydesign.html?highlight=scope#scoped-based-timer), which is usually ad hoc and does not have a lot of simple examples online.  We may use the examples of the sample-based profiling, but start with a scope-based one.

Reply:
Thanks for the comments. The roadmap sounds good to me. Let me have a look of  https://github.com/yse/easy_profiler